tests/lib/openstack_query/test_query_methods.py

```diff
@@ -145,12 +148,25 @@ def test_run_with_parsing_grouped(self):
         Tests that run method works expectedly - with parsing - i.e. grouping into dict
         Should call run_parser to get items - either dict of lists or list and handle them properly
         """
+        mock_client_filter_func = self.mock_builder.client_side_filter
+        mock_server_filters = self.mock_builder.server_side_filters
         self.mock_parser.run_parser.return_value = ["obj1", "obj2"]
         self.mock_parser.run_parser.return_value = {
             "group1": ["obj1", "obj2"],
             "group2": ["obj3", "obj4"],
         }
-        self.instance.run("cloud_account")
+
+        # should implicitly convert it to a valid enum in the function
+        self.instance.run("DEV")
+
+        self.mock_runner.run.assert_called_once_with(
+            "dev", mock_client_filter_func, mock_server_filters, None
+        )
+
+        self.mock_parser.run_parser.assert_called_once_with(
+            self.mock_runner.run.return_value
+        )
```

Comment:
I think we can drop this, since the prod case shows that it works whilst this exercises the same code path anyway?

Reply:
No - this test checks that `run` can handle when the results returned by `run_parser` are grouped - i.e the method returns a `dict` rather than a list